sklearn/inspection/_partial_dependence.py

```diff
@@ -349,21 +367,34 @@ def partial_dependence(estimator, X, features, response_method='auto',
                 "'decision_function'. Got {}.".format(response_method)
             )
 
-    n_features = X.shape[1]
-    features = np.asarray(features, dtype=np.int32, order='C').ravel()
-    if any(not (0 <= f < n_features) for f in features):
-        raise ValueError('all features must be in [0, %d]'
-                         % (n_features - 1))
+    if _determine_key_type(features) == 'int':
+        if (isinstance(features, Iterable) or
```

Comment:
If `_determine_key_type` returns 'int' is this check for `Iterable` needed?

Reply:
Specifically, if `_determine_key_type(features) == 'int'`, when is the following false?

```py
isinstance(features, Iterable) or isinstance(features, numbers.Integral)
```

Reply:
`slice(0, 10)` which is supported.

Reply:
We should not support slices

Reply:
OK so there are 2 possibilities here:

1. raise an error specifically for a `slice`, otherwise we would get a weird error.
2. or just silently make it works (with the current `if` statement and we already have a test)

Reply:
I also think we should not support slices (sounds like a yagni) and make sure that the error message is helpful enough if the user pass something that cannot be interpreted as a valid (set of) integer indices for numpy data or a (set of) column names for pandas dataframes.